packages/components/src/internal/components/samples/SampleStatusLegend.tsx

```diff
@@ -0,0 +1,62 @@
+import React, {FC, memo} from "react";
+
+import {SampleStatusTag} from "./SampleStatusTag";
+import {getSampleStatus} from "./utils";
+import {SCHEMAS} from "../../schemas";
+import {InjectedQueryModels, withQueryModels} from "../../../public/QueryModel/withQueryModels";
+import {QuerySort} from "../../../public/QuerySort";
+import {isLoading} from "../../../public/LoadingState";
+import {LoadingSpinner} from "../base/LoadingSpinner";
+import {caseInsensitive} from "../../util/utils";
+
+interface OwnProps {}
+
+const SampleStatusLegendImpl: FC<OwnProps & InjectedQueryModels> = memo(props => {
+    const { model } = props.queryModels;
+
+    if (isLoading(model.rowsLoadingState)) {
+        return <LoadingSpinner />;
+    }
+
+    if (model.rowCount === 0) {
+        return (
+            <table className="sample-status-legend--table">
+                <tr>
```

Comment:
Technically,  there should be a `<tbody>` before the `<tr>`.

Reply:
Thanks. I added the `tbody` for the "has sample statuses" case but forgot it here. Done.